pkg/security/ebpf/c/raw_syscalls.h

```diff
@@ -53,31 +44,103 @@ int sys_enter(struct _tracepoint_raw_syscalls_sys_enter *args) {
     bpf_probe_read(&syscall.id, sizeof(syscall.id), &args->id);
     bpf_get_current_comm(&syscall.comm, sizeof(syscall.comm));
 
-    u32 zero = 0;
-    u32 *back_buffer;
-    if (!(back_buffer = bpf_map_lookup_elem(&noisy_processes_buffer, (void*) &zero)))
+    u32 selector_key = SYSCALL_MONITOR_KEY;
+    u32 *buffer_id;
+    if (!(buffer_id = bpf_map_lookup_elem(&buffer_selector, &selector_key)))
         return 0;
 
-    u64 count = 0;
-    u64 *current_count;
-
-    if (*back_buffer) {
-        current_count = bpf_map_lookup_elem(&noisy_processes_bb, &syscall);
+    u64 *count;
+    if (*buffer_id) {
+        count = bpf_map_lookup_elem(&noisy_processes_bb, &syscall);
     } else {
-        current_count = bpf_map_lookup_elem(&noisy_processes_fb, &syscall);
+        count = bpf_map_lookup_elem(&noisy_processes_fb, &syscall);
     }
 
-    if (current_count) {
-        count = *current_count;
+    if (count) {
+        __sync_fetch_and_add(count, 1);
+    } else {
+        u64 one = 1;
+        count = &one;
+
+        if (*buffer_id) {
+            bpf_map_update_elem(&noisy_processes_bb, &syscall, count, BPF_ANY);
+        } else {
+            bpf_map_update_elem(&noisy_processes_fb, &syscall, count, BPF_ANY);
+        }
     }
-    count++;
 
-    if (*back_buffer) {
-        bpf_map_update_elem(&noisy_processes_bb, &syscall, &count, BPF_ANY);
+    return 0;
+}
```

Comment:
Thank you @L3n41c ! I was afraid that passing map pointers in a function would fail since we tried it before and the verifier didn't seem to like it. However the kitchen tests seem to work fine on all the major distributions that we support, which means that we should be all good !